TrackModel/src/trackmodel/model/Block.java

```diff
@@ -230,44 +196,71 @@ public boolean getHeaters() {
     return heaters;
   }
 
+
   /**
-   * The status of the switch id number will be returned if this block holds a switch.
-   * @return The integer that is used as an id will be returned
+   * Return the id of any train present.
+   * @return An integer identification for a train is passed back
    */
-  public int getSwitchId() {
-    return switchId;
+  public int getTrainPresent() {
+    return trainPresent;
   }
 
   /**
-   * The switch at the current location will be returned.
-   * @return Returns a switch object
-   */
-  public Switch getSwitch() {
-    return junction;
+  * This method will alert the driver if the left doors are to be opened.
+  * @return A Boolean value indicating if the station is on the left
+  */
+  public boolean getLeftDoors() {
+    if (stationName.equals("")) {
+      return leftStation;
+    }
+    return false;
   }
 
   /**
-   * The switch will return a crossing object if one is at the block.
-   * @return Returns a crossing object
-   */
-  public Crossing getCrossing() {
-    return crossing;
+  * This method will alert the driver if the left doors are to be opened.
+  * @return A Boolean value indicating if the station is on the right
+  */
+  public boolean getRightDoors() {
+    if (stationName.equals("")) {
+      return rightStation;
+    }
+    return false;

```

Comment:
If stationName is empty, then return rightStation. Why?

Reply:
This will indicate if there are limitations of doors that can open.  If the station is on the right side of the track let the train know so that they can figure out which doors to open.

Reply:
Never mind I meant to flip that that should be !stationName.equal("")

Reply:
This is still an issue. You should return null if there is no station, not false.